### PR TITLE
Fix -Wall and -Wunused-packages in stan plugin

### PR DESCRIPTION
**Pull request by jhrcek**
This PR is part of larger effort where I try to reduce the amount of compiler warnings across all packages.

### CODE REVIEW

File: plugins/hls-stan-plugin/src/Ide/Plugin/Stan.hs [hunk 1]
```diff
@@ -162,24 +139,21 @@ rules recorder plId = do
 
               -- If envVar is set to 'False', stan will ignore all local and global .stan.toml files
               logWith recorder Debug (LogDebugStanEnvVars env)
-              seTomlFiles <- liftIO $ usedTomlFiles useDefConfig (stanArgsConfigFile stanArgs)
```
Comment (jhrcek) [inline plugins/hls-stan-plugin/src/Ide/Plugin/Stan.hs:165]: I'm not sure about this, but seTomlFiles unused past this point.
There's also identical statement executed 5 lines above this and the seTomlFiles are just logged.

File: plugins/hls-stan-plugin/src/Ide/Plugin/Stan.hs [hunk 2]
```diff
@@ -162,24 +139,21 @@ rules recorder plId = do
...
+                  ResultL _warnings stanConfig -> do
                       let currentHSAbs = fromNormalizedFilePath file -- hie_hs_file hie
                       currentHSRel <- liftIO $ makeRelativeToCurrentDirectory currentHSAbs
                       cabalExtensionsMap <- liftIO $ createCabalExtensionsMap isLoud (stanArgsCabalFilePath stanArgs) [hie]
 
                       -- Files (keys) in checksMap need to have an absolute path for the analysis, but applyConfig needs to receive relative
                       -- filepaths to apply the config, because the toml config has relative paths. Stan itself seems to work only in terms of relative paths.
                       let checksMap = HM.mapKeys (const currentHSAbs) $ applyConfig [currentHSRel] stanConfig
-
-                      let analysis = runAnalysis cabalExtensionsMap checksMap (configIgnored stanConfig) [hie]
```
Comment (jhrcek) [inline plugins/hls-stan-plugin/src/Ide/Plugin/Stan.hs:182]: Not sure if this should be removed, but `analysis` - result of this pure computation is unused below, so seems pointless to compute.

**Comment by michaelpj** [comment]
Sounds like there are some genuine questions, perhaps @0rphee can advise.

File: plugins/hls-stan-plugin/src/Ide/Plugin/Stan.hs (hunk 2 above)
Comment (0rphee) [reply plugins/hls-stan-plugin/src/Ide/Plugin/Stan.hs:182]: Yes, it seems I forgot to delete that line when iterating with this function.

File: plugins/hls-stan-plugin/src/Ide/Plugin/Stan.hs (hunk 1 above)
Comment (0rphee) [reply plugins/hls-stan-plugin/src/Ide/Plugin/Stan.hs:165]: This also slipped past me. The precious (repeated) statement is used to log which .stan.toml files were used to load the stan config. 

The binding should have a better name though. I think `stanTomlUsed` may be more useful.

**Comment by 0rphee** [comment]
Everything seems fine. Some errors seem to have eluded me!

File: plugins/hls-stan-plugin/src/Ide/Plugin/Stan.hs (hunk 1 above)
Comment (jhrcek) [reply plugins/hls-stan-plugin/src/Ide/Plugin/Stan.hs:165]: Ok, I renamed the other binding to `tomlsUsedByStan`

**Comment by jhrcek** [comment]
@0rphee thanks for checking. @michaelpj This PR should be merge-ready.

**Review by 0rphee — APPROVED** [review]

**Pull request merged**